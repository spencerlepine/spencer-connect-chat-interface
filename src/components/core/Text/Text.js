// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import React, { PureComponent } from 'react';
import PT from 'prop-types';
import styled from 'styled-components';

const StyledH1 = styled.h1`
  ${({ styleAs, theme }) => (theme.typography[styleAs || 'h1'])}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
  ${({ inline }) => inline && 'display:inline;'}
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.h1Margin : 0)};
`;

const StyledH2 = styled.h2`
  ${({ styleAs, theme }) => (theme.typography[styleAs || 'h2'])}
  ${({ inline }) => inline && 'display:inline;'}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.h2Margin : 0)};
`;
const StyledH3 = styled.h3`
  ${({ styleAs, theme }) => (theme.typography[styleAs || 'h3'])}
  ${({ inline }) => inline && 'display:inline;'}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.h3Margin : 0)};
`;
const StyledH4 = styled.h4`
  ${({ styleAs, theme }) => (theme.typography[styleAs || 'h4'])}
  ${({ inline }) => inline && 'display:inline;'}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.h4Margin : 0)};
`;
const StyledH5 = styled.h5`
  ${({ styleAs, theme }) => (theme.typography[styleAs || 'h5'])}
  ${({ inline }) => inline && 'display:inline;'}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.h5Margin : 0)};
`;
const StyledH6 = styled.h6`
  ${({ styleAs, theme }) => (theme.typography[styleAs || 'h6'])}
  ${({ inline }) => inline && 'display:inline;'}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.h6Margin : 0)};
`;

const StyledP = styled.p`
  ${({ theme }) => theme.typography.base}
  ${({ inline }) => inline && 'display:inline;'}
  color: ${({ theme, textColor }) => (theme.typography.color[textColor] || 'inherit')};
 
  margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.baseMargin : 0)};
 
  &:last-child {
    margin: ${({ theme: { typography }, withMargin }) => (withMargin ? typography.baseMargin : 0)};
  }
`;
export default class Text extends PureComponent {
  static propTypes = {
    type: PT.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p']),
    withMargin: PT.bool,
    styleAs: PT.string,
    inline: PT.bool,
    textColor: PT.oneOf(['base', 'secondary', 'tertiary']),
  }

  static defaultProps = {
    type: 'p',
    withMargin: false,
    styleAs: undefined,
    inline: false,
    textColor: null,
  }

  render() {
    const {
      type,
      ...rest
    } = this.props;

    switch (type) {
      case 'h1': return <StyledH1 {...rest} />;
      case 'h2': return <StyledH2 {...rest} />;
      case 'h3': return <StyledH3 {...rest} />;
      case 'h4': return <StyledH4 {...rest} />;
      case 'h5': return <StyledH5 {...rest} />;
      case 'h6': return <StyledH6 {...rest} />;
      case 'p': return <StyledP {...rest} />;
      default: return <StyledP {...rest} />;
    }
  }
}