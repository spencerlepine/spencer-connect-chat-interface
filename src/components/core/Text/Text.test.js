// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import React from 'react';
import Text from './';

describe('<Text />', () => {
  describe('styles should match snapshots: ', () => {
    test('h1', () => {
      const tree = createTree(<Text type="h1">H1</Text>);
      expect(tree).toMatchSnapshot();
    });

    test('h2', () => {
      const tree = createTree(<Text type="h2">H2</Text>);
      expect(tree).toMatchSnapshot();
    });

    test('h3', () => {
      const tree = createTree(<Text type="h3">H3</Text>);
      expect(tree).toMatchSnapshot();
    });

    test('h4', () => {
      const tree = createTree(<Text type="h4">H4</Text>);
      expect(tree).toMatchSnapshot();
    });

    test('h5', () => {
      const tree = createTree(<Text type="h5">H5</Text>);
      expect(tree).toMatchSnapshot();
    });

    test('h6', () => {
      const tree = createTree(<Text type="h6">H6</Text>);
      expect(tree).toMatchSnapshot();
    });

    test('p', () => {
      const tree = createTree(<Text type="p">P</Text>);
      expect(tree).toMatchSnapshot();
    });
  });

  it('can style a tag as something else', () => {
    const tree = createTree(<Text type="h1" styleAs="h2">H1 tag that looks like H2</Text>);
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-size', '21px');
  });

  it('can style inline', () => {
    const tree = createTree(<Text inline>I am inline</Text>);
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('display', 'inline');
  });
});