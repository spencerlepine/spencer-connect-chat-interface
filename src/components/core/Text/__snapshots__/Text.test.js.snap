// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> can style a tag as something else 1`] = `
.c0 {
  font-size: 21px;
  line-height: 1.3;
  font-weight: 700;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h1
  className="c0"
>
  H1 tag that looks like H2
</h1>
`;

exports[`<Text /> can style inline 1`] = `
.c0 {
  font-size: 1em;
  line-height: 1.465;
  font-weight: normal;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  display: inline;
  color: inherit;
  margin: 0;
}

.c0:last-child {
  margin: 0;
}

<p
  className="c0"
>
  I am inline
</p>
`;

exports[`<Text /> styles should match snapshots:  h1 1`] = `
.c0 {
  font-size: 28px;
  line-height: 1.2;
  font-weight: normal;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h1
  className="c0"
>
  H1
</h1>
`;

exports[`<Text /> styles should match snapshots:  h2 1`] = `
.c0 {
  font-size: 21px;
  line-height: 1.3;
  font-weight: 700;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h2
  className="c0"
>
  H2
</h2>
`;

exports[`<Text /> styles should match snapshots:  h3 1`] = `
.c0 {
  font-size: 17px;
  line-height: 1.255;
  font-weight: 700;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h3
  className="c0"
>
  H3
</h3>
`;

exports[`<Text /> styles should match snapshots:  h4 1`] = `
.c0 {
  font-size: 17px;
  line-height: 1.255;
  font-weight: normal;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h4
  className="c0"
>
  H4
</h4>
`;

exports[`<Text /> styles should match snapshots:  h5 1`] = `
.c0 {
  font-size: 13px;
  line-height: 1.465;
  font-weight: 700;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h5
  className="c0"
>
  H5
</h5>
`;

exports[`<Text /> styles should match snapshots:  h6 1`] = `
.c0 {
  font-size: 11px;
  line-height: 1.465;
  font-weight: normal;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  text-transform: uppercase;
  padding: 0;
  color: inherit;
  margin: 0;
}

<h6
  className="c0"
>
  H6
</h6>
`;

exports[`<Text /> styles should match snapshots:  p 1`] = `
.c0 {
  font-size: 1em;
  line-height: 1.465;
  font-weight: normal;
  font-family: Arial,Helvetica,"sans-serif",Helvetica,sans-serif;
  padding: 0;
  color: inherit;
  margin: 0;
}

.c0:last-child {
  margin: 0;
}

<p
  className="c0"
>
  P
</p>
`;
